src/main/java/week6/ParkingLot.java

```diff
@@ -0,0 +1,152 @@
+package week6;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.logging.Logger;
+
+public class ParkingLot {
+
+  private static final int EMPTY = 0;
+  private static final Logger logger = Logger.getLogger(ParkingLot.class.getName());
+
+  /*
+   * First, we create a map of wrongly positioned cars, car no. -> position
+   * Iteratively, replace 0 with the correct car. We remove cars that are in correct pos from map
+   *
+   * If 0 becomes in the correct position but map is not empty, move 0 to first incorrect car pos
+   * and start iterating again until map is empty.
+   *
+   * Amortized Time Complexity: O(n)
+   * Space Complexity: O(n)
+   *
+   * where n is the size of the array source
+   *
+   * Assumptions:
+   *    1) len(source) = len(target)
+   *    2) arrays contain O...n, this means no duplications
+   *
+   * @param source : the initial configuration of the cars
+   * @param target : the target configuration of the cars
+   * @return : List of Moves to be performed to transform source -> target
+   */
+  public static List<Move> rearrangeCars(final int[] source, final int[] target) {
+
+    List<Move> moves = new ArrayList<Move>() {
+      @Override
+      public String toString() {
+        return super.toString().replaceAll("\\[|]|(, )", "");
+      }
+    };
+
+    if (source == null || target == null || source.length != target.length) {
+      return moves;
+    }
```

Comment:
"source.length == target.length" was a documented *assumption*, but you check it anyway. Are you assuming the input is valid, or not?

Secondly, returning an empty list of moves is a horrible way to signal an error. Inputs that are already in the correct position can quite validly return an empty list. It would be better to return null, but far better in my opinion to throw an exception. This is what exceptions are for - exceptional conditions, and people passing in totally invalid inputs are a good situation in which to throw exceptions.